symengine/subs.h

```diff
@@ -192,7 +192,39 @@ class XReplaceVisitor : public BaseVisitor<XReplaceVisitor>
                 throw SymEngineException("expected an object of type Boolean");
             v.insert(rcp_static_cast<const Boolean>(a));
         }
-        result_ = x.create(v);
+        result_ = logical_and(v);
+    }
+
+    void bvisit(const Or &x)
+    {
+        set_boolean v;
+        for (const auto &elem : x.get_container()) {
+            auto a = apply(elem);
+            if (not is_a_Boolean(*a))
+                throw SymEngineException("expected an object of type Boolean");
+            v.insert(rcp_static_cast<const Boolean>(a));
+        }
+        result_ = logical_or(v);
+    }
+
+    void bvisit(const Not &x)
```

Comment:
It seems that `Not` gets automatically simplified away in the expressions, so this method is never called - is that always the case?

Reply:
Good catch - looks to me like Not class is never actually instantiated in framework. I'd think if class remains in framework it should be handled correctly in subs but this probably argues for just removing the class entirely?

Reply:
What do you mean?  `not(x)` will be using the `Not` class

Reply:
@rikardn yes, but since `Not` can currently only be constructed from a `Boolean`, it always gets simplified away, e.g. `Not(x>5)` becomes `x<=5`, so this `bvisit(const Not &x)` never gets called.

@tbb300 I agree, and currently it does look like `Not` could be removed entirely, but I guess the idea is to support things like `Not(x)` in the future, so probably as above it makes sense to keep your PR as it is (even though we can't currently write a test for these lines)

Reply:
Ok, I didn't know, but you can still have `logical_not(logical_or(...))`

Reply:
@rikardn the `Not` still gets simplified away, e.g. in the test case in this PR:
`logical_not(Or(1 < x, x < 3))` -> `And(3 <= x, x <= 1)`

Reply:
Here's an example where `Not` is not simplified away. https://github.com/symengine/symengine/blob/master/symengine/tests/logic/test_logic.cpp#L234-L249

Reply:
Thanks @isuruf for saving me :-)   You could also have `logical_not(logical_xor(...))`